### ISSUE TITLE
Build error: Property 'name' is missing

### ISSUE THREAD

**Issue by jeengbe** [lukeautry/tsoa#1582]
## Sorting

- **I'm submitting a ...**

  - [x] bug report
  - [ ] feature request
  - [ ] support request

- I confirm that I
  - [x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior

should build successfully 

## Current Behavior

```
src/build/tsoa/routes.ts:37:21 - error TS2741: Property 'name' is missing in type '{ in: string; required: true; dataType: "object"; }' but required in type 'ParameterSchema'.

37                     undefined: {"in":"request","required":true,"dataType":"object"},
                       ~~~~~~~~~

  ../../node_modules/.pnpm/@tsoa+runtime@6.1.1/node_modules/@tsoa/runtime/dist/routeGeneration/tsoa-route.d.ts:50:9
    50         name: string;
               ~~~~
    'name' is declared here.


Found 1 error in src/build/tsoa/routes.ts:37
```

## Steps to Reproduce

```ts
import type { Request as KoaRequest } from 'koa';

  handleRequest(
    @Request() { req }: KoaRequest,
  ) {}	
```

## Context (Environment)

Version of the library: 6.1.1

- Confirm you were using yarn not npm: [x]

**Comment by jackey8616** [comment]
@jeengbe 
Can you provides the version of 'koa' what you use plz.

Since that might cause by `req` field is not found in KoaRequest.

**Comment by jeengbe** [comment]
Sorry, I forgot to add this code snippet; this is what tsoa generates:

```ts
            const args: Record<string, TsoaRoute.ParameterSchema> = {
                    undefined: {"in":"request","required":true,"dataType":"object"},
            };
```

I assume this is because the parameter has no name, instead, is deconstructed directly. This issue was introduced with tsoa 6.1 and works fine on 6.0.

**Comment by jackey8616** [comment]
@jeengbe 
I dig a little bit into the code.

This issue is not directly cause by recently update(^6.0.1), but discovered after implemented TemplateService to guard .hbs templates types.

The `@Request` is not support any deconstructing function parameter from the begining.
(You can check the routes.ts generate by v6.0.1, you will find out that the key in args is still undefined)

TL;DR
Fortunately, we do have a new decorator named `@RequestProp` to let you extract field from request, but currently we are not supporting any out-source type. So you have to do something like this:
```
import { IncomingMessage } from 'http';

handleRequest(
    @RequestProp('req') req: any,
) {
    // IncomingMessage is the type defined in koa's Request req filed.
    const koaReq: IncomingMessage = req as IncomingMessage;
    // other stuff you want
}
```

**Comment by WoH** [comment]
Did this ever work before? Or just not throw an error on build.

**Comment by shibukazu** [comment]
Is this planned to be corrected in the future? Or will it remain as is?

**Comment by jackey8616** [comment]
> Is this planned to be corrected in the future? Or will it remain as is?

This is not an issue, you can consider use `RequestProp` to achieve similar functionality.
